### PR TITLE
Instead of generating expected JUnit output, record it

### PR DESCRIPTION
**Pull request by gzm0**

### CODE REVIEW

File: junit-runtime/src/main/scala/org/scalajs/junit/JUnitExecuteTest.scala [hunk 1]
```diff
@@ -161,7 +161,7 @@ final class JUnitExecuteTest(task: JUnitTask, runSettings: RunSettings,
   }
 
   private def emitMethodEvent(methodName: String, status: Status): Unit = {
-    val selector = new NestedTestSelector(task.taskDef.fullyQualifiedName, methodName)
+    val selector = new TestSelector(task.taskDef.fullyQualifiedName + "." + runSettings.decodeName(methodName))
```
Comment (sjrd) [inline junit-runtime/src/main/scala/org/scalajs/junit/JUnitExecuteTest.scala]: Is that a bug fix? If so, it would be good if it could have its own commit message, or at least be mentioned in the commit message.

File: junit-test/shared/src/test/scala/org/scalajs/junit/utils/JUnitTest.scala [hunk 2]
```diff
@@ -12,401 +12,234 @@
...
-
   @Test def testJUnitOutput(): Unit = {
-    for (frameworkArgs <- frameworkArgss) {
-      val jUnitFramework = new com.novocode.junit.JUnitFramework()
-      val runner =
-        jUnitFramework.runner(frameworkArgs.toArray, Array.empty, classLoader)
-      val tasks = runner.tasks(Array(new TaskDef(suiteUnderTestName,
-        jUnitFramework.fingerprints.head, true, Array.empty)))
+    for (args <- frameworkArgss) {
+      val rawOut = runTests(args.map("-" + _), new JUnitTestRecorder)
```
[the remark this answers is not in the document]
Comment (gzm0) [reply junit-test/shared/src/test/scala/org/scalajs/junit/utils/JUnitTest.scala]: Right, I was going to give it arguments in a later PR, but even then it is better like this.

File: junit-test/shared/src/test/scala/org/scalajs/junit/utils/JUnitTest.scala [hunk 3]
```diff
@@ -12,401 +12,234 @@
...
-        .matchOne(List(testRunFinishedOutput(total, ignored, failed), done))
-        .assertEmpty()
-    }
-
-    // Text builders.
 
-    private def done: Output = Done("")
+    case Event(_, name) =>
+      val n = name.stripPrefix(testNamePrefix)
+      if (n == "") None else Some(n)
```
[the remark this answers is not in the document]
Comment (gzm0) [reply junit-test/shared/src/test/scala/org/scalajs/junit/utils/JUnitTest.scala]: Yeah, this is wrong. It should do:

```
"foo.BarTest.test1" -> Some("test1")
"foo.BarTest" -> None
```

**Comment by gzm0** [comment]
Updated.

**Comment by gzm0** [comment]
Updated: made the test name detection more restrictive.
